.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,108 @@
+#   automatically deploy the new results to
+#   `nilearn.github.io/benchmarks <https://nilearn.github.io/benchmarks/>`_.
+###
+name: Run and deploy Nilearn benchmarks
+
+on:
+    push:
+        branches:
+        -   main
+    schedule:
+    # everday at 10pm UTC
+    -   cron: 0 22 * * *
+
+jobs:
+    benchmark:
+        if: >-
+            (contains(github.event.head_commit.message, '[bm]') ||
+            github.event_name == 'schedule') &&
+            github.repository == 'nilearn/nilearn'
+        # TODO bump to 24.04 when bumping minimum python to 3.10
+        runs-on: ubuntu-22.04
+        steps:
+        -   uses: actions/checkout@v3
+        -   uses: actions/setup-python@v3
+        -   name: Install asv
+            run: |
+                pip install --upgrade pip
+                pip install asv
+        -   name: Add SSH key for benchmarks repo
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent_benchmarks.sock
+            run: |
+                mkdir -p ~/.ssh
+                ssh-keyscan github.com >> ~/.ssh/known_hosts
+                echo "${{ secrets.UPLOAD_BENCHMARK_RESULTS }}" > ~/.ssh/github_actions
+                chmod 600 ~/.ssh/github_actions
+                ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+                ssh-add ~/.ssh/github_actions
+        -   name: Pull previous results from benchmarks repo
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent_benchmarks.sock
+            run: |
+                git clone git@github.com:nilearn/benchmarks.git benchmarks_repo
+                cp -r benchmarks_repo/results .
+        -   name: Get all the machine info
+            run: |
+                asv machine --yes
+        -   name: Edit asv-machine.json to a custom machine name
+            run: python ./build_tools/github/set_machine_name.py fv-az1113-357
+        -   name: Run all benchmarks on the latest commit
+            run: |
+                asv run -ev --append-samples --machine fv-az1113-357
+        -   name: Create html with all results
+            run: |
+                asv publish
+        -   uses: actions/upload-artifact@v4
+            with:
+                name: Upload asv benchmark results as artifacts
+                path: |
+                    ./env
+                    ./html
+                    ./results
+                compression-level: 9
+        -   name: Push new results back to benchmarks repo and deploy
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+            run: |
+                cd benchmarks_repo
+                cp -r ../results .
+                cp -r ../html/* .
+                git config --global user.name "GitHub Actions"
+                git config --global user.email "actions@github.com"
+                git add .
+                git commit -m "Update benchmark results and HTML"
+                git push origin main
+                cd ..
```

Comment:
are we sure we always want to do this?

say we trigger the workflow on a PR, then we would update our benchmark results repo with results from a PR.

I think we want the benchmark repo to only have things from the main branch, no?

Reply:
As of now, this workflow is only for the main, it can't be triggered from PR. 
We decided to hold off on running benchmarks from PRs.

If someone wants to compare with their work on a PR it would have to be done locally. I could add more info on that in the doc but I haven't tried that myself yet.

Reply:
ha yeah but then it feels weird to have the workflow "triggerable" by commit message on main

ok let's keep thing as is for now and then get back to this when we allow to run this on PR

Reply:
Oh, you mean one could potentially name their PR like "[bm] ...." and that would trigger this (since we squash our PR commits). I did not think about that tbh.

I was simply thinking of doing an empty commit with [bm] on the main to trigger it...